File: practice23/6.cpp


//https://www.nowcoder.com/practice/57260c08eaa44feababd05b328b897d7?tpId=98&tqId=32880&tPage=3&rp=3&ru=/ta/2019test&qru=/ta/2019test/question-ranking
#include<iostream>
#include<string>
using namespace std;

void bracket_matching(string& str)
{
	char *ch = new char[str.size()];
	int m = 0;
	for (int i = 0; i < str.size(); i++)
	{
		if (m >= 1 && str[i] == ')' && ch[m - 1] == '(')
		{
			m--;
		}
		else if (m >= 1 && str[i] == ']' && ch[m - 1] == '[')
		{
			m--;
		}
		else if (str[i] == '(' || str[i] == ')' || str[i] == '[' || str[i] == ']')
		{
			ch[m] = str[i];
			m++;
		}
	}
	if (m == 0)
	{
		cout << "true" << endl;
	}
	else
	{
		cout << "false" << endl;
	}
	delete[]ch;
}

int main()
{
	string str;
	while (cin >> str)
	{
		bracket_matching(str);
	}
	system("pause");
	return 0;
}

File: practice32/6.java



//https://www.nowcoder.com/practice/74475ee28edb497c8aa4f8c370f08c30?tpId=85&tqId=29836&tPage=1&rp=1&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s1=in.next();
        String s2=in.next();
        StringBuilder s=new StringBuilder(s1);
        StringBuilder t=new StringBuilder(s2);
        int x=0;
        char ch=t.charAt(x);
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)==ch){
                x++;
                if(x==t.length()){
                    break;
                }
                ch=t.charAt(x);
            }
        }
        if(x==t.length()){
            System.out.println("Yes");
        }
        else{
            System.out.println("No");
        }
    }
}